resources/components/App.vue

```diff
@@ -193,11 +193,9 @@ module.exports = {
 		// Listen for hash changes.
 		window.addEventListener( 'hashchange', this.onHashChange );
 
-		// popular images are pre-loaded on the server side;
-		// immediately fetch user images in the mounted hook so that they'll be
-		// ready for the user if they switch tabs
+		// Fetch batch of images.
 		this.getImages( {
-			queue: 'user'
+			queue: hash
```

Comment:
Previously, we relied on the popular images to be fetched on the server and fetched the user images here in the background.

Now we are only fetching one of the two queues – whichever one matches the hash in the URL when the user first lands here. The watcher that fires off new requests only does so if the number of images in a queue decreases from 1 to 0; if there were never images in the first place then more won't be fetched.

I think the solution here would simply to be to have `this.getImages` run twice, once for each queue. Ideally we'd want to populate both queues as soon as possible regardless of which tab the user starts on, so that we're prepared for when they switch.

Reply:
Good call, I pushed a commit making that update!